test/sql/include/append.sql

```diff
@@ -111,6 +109,15 @@ WHERE time > now_s() - interval '4 months'
 GROUP BY t
 ORDER BY t DESC;
 
+-- querying outside the time range should return nothing
+EXPLAIN (costs off)
```

Comment:
Two things: 1) Shouldn't we execute this as well as EXPLAIN it? 2) In the unoptimized version of this plan I don't see the Result node replacing the append node. How is the case with the replacement hit? Am I missing something?

Reply:
So, what is happening here, I believe, is that constraint exclusion will exclude all chunks based on the `WHERE` clause on time, while `date_part` enables `ConstraintAwareAppend` This happens in both the optimized and unoptimized case. This leaves the main table in the Append plan. With `ConstraintAwareAppend` we remove the main table from the plan, since it holds no tuples. Since this, together with constraint exclusion, removes all child tables from the Append path node, it will be turned into a `Result` node at planning time. The unoptimized case does not remove the main table from the Append, since it has no constraints, and it will therefore not replace the plan with a Result node.

Here's the relevant code in Postgres:

https://github.com/postgres/postgres/blob/0f2458ff5f970cade04313f1a10fe01d02f888b7/src/backend/optimizer/plan/createplan.c#L1028